ebms-provider/src/main/kotlin/no/nav/emottak/ebms/ebxml/ApplicationCallExt.kt

```diff
@@ -0,0 +1,191 @@
+package no.nav.emottak.ebms.ebxml
+
+import io.ktor.client.request.forms.MultiPartFormDataContent
+import io.ktor.http.ContentType
+import io.ktor.http.HeadersBuilder
+import io.ktor.http.HttpStatusCode
+import io.ktor.http.content.PartData
+import io.ktor.http.content.forEachPart
+import io.ktor.http.content.streamProvider
+import io.ktor.server.application.ApplicationCall
+import io.ktor.server.request.contentType
+import io.ktor.server.request.header
+import io.ktor.server.request.receive
+import io.ktor.server.request.receiveMultipart
+import io.ktor.server.response.respond
+import io.ktor.server.response.respondText
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import no.nav.emottak.constants.SMTPHeaders
+import no.nav.emottak.ebms.log
+import no.nav.emottak.ebms.model.DokumentType
+import no.nav.emottak.ebms.model.EbMSDocument
+import no.nav.emottak.ebms.validation.MimeHeaders
+import no.nav.emottak.ebms.validation.MimeValidationException
+import no.nav.emottak.ebms.validation.validateMimeAttachment
+import no.nav.emottak.ebms.validation.validateMimeSoapEnvelope
+import no.nav.emottak.ebms.xml.asByteArray
+import no.nav.emottak.ebms.xml.asString
+import no.nav.emottak.ebms.xml.getDocumentBuilder
+import no.nav.emottak.melding.model.EbmsAttachment
+import no.nav.emottak.util.createUniqueMimeMessageId
+import java.io.ByteArrayInputStream
+import java.nio.charset.StandardCharsets
+import java.util.Base64
+import java.util.UUID
+
+@Throws(MimeValidationException::class)
+suspend fun ApplicationCall.receiveEbmsDokument(): EbMSDocument {
+    log.info("Parsing message with Message-Id: ${request.header(SMTPHeaders.MESSAGE_ID)}")
+
+    return when (val contentType = this.request.contentType().withoutParameters()) {
+        ContentType.parse("multipart/related") -> handleMultipartRelated()
+        ContentType.parse("text/xml") -> handleTextXml()
+        else -> {
+            throw MimeValidationException("Ukjent request body med Content-Type $contentType")
+        }
+    }
+}
+
+suspend fun ApplicationCall.handleTextXml(): EbMSDocument {
```

Comment:
Jeg vet ikke hva er riktin navn for denne methoden. Kan stå sånn for øyeblikket.